### PR TITLE
Add a cleanup job for Azure Eventhubs Namespaces

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

This is a periodic job to cleanup Azure Eventhubs Namespaces that exceeded the maximum TTL.

**Related issue(s)**
- Fixes https://github.com/kyma-project/test-infra/issues/2797.

### CODE REVIEW

File: templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - <<: *base_image_116
+        securityContext:
+          privileged: true
+        command:
+          - "bash"
+        args:
+          - "-c"
+          - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/cluster-integration/helpers/cleanup-azure-event-hubs-namespaces.sh"
+        env:
+          - name: AZURE_SUBSCRIPTION_APP_ID
```
Comment (sayanh) [inline templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml:29]: There is already a [preset](https://github.com/kyma-project/test-infra/blob/28bdc7b7b9d0069dcba4455aa4462bc803b071e8/prow/config.yaml#L197) meant for this purpose.

One example of using this is [here](https://github.com/kyma-project/test-infra/blob/master/prow/jobs/kyma/kyma-upgrade-gardener.yaml#L17).

File: templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      testgrid-create-test-group: "false"
+    <<: *azure_event_hubs_namespaces_cleanup_job_template
+    cron: "30 * * * *" # "At minute 30 past every hour"
+    decoration_config:
+      timeout: 1800000000000 # 30min
+      grace_period: 600000000000 # 10min
+    extra_refs:
+      - <<: *test_infra_ref
+        base_ref: master
+      - <<: *kyma_ref
```
Comment (sayanh) [inline templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml:70]: Why do we need to clone Kyma repo?

File: prow/scripts/cluster-integration/helpers/cleanup-azure-event-hubs-namespaces.sh [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+#   AZURE_SUBSCRIPTION_ID
+# ARGUMENTS:
+#   None
+# OUTPUTS:
+#   Print the result of authenticating to Azure.
+# RETURN:
+#   0 if success, non-zero on error.
+#########################################################################################################
+function authenticate_to_azure() {
+  echo "#################################################################################################"
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply prow/scripts/cluster-integration/helpers/cleanup-azure-event-hubs-namespaces.sh]: Done.

File: prow/scripts/cluster-integration/helpers/cleanup-azure-event-hubs-namespaces.sh [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  AZURE_SUBSCRIPTION_ID
+  AZURE_SUBSCRIPTION_NAME
+  TTL_HOURS
+)
+
+#########################################################################################################
+# Constants
+#########################################################################################################
+readonly NOW="$(date +%s)"     # The current timestamp when the script runs.
+readonly SECONDS_PER_HOUR=3600 # The amount of seconds per minute.
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply prow/scripts/cluster-integration/helpers/cleanup-azure-event-hubs-namespaces.sh]: Done.

File: templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml (hunk 1 above)
Comment (marcobebway) [reply templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml]: Done.

File: templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml (hunk 2 above)
Comment (marcobebway) [reply templates/templates/kyma-azure-event-hubs-namespaces-cleanup.yaml]: We don't need it.

Removed.

**Review by sayanh — APPROVED** [review]

**Review by k15r — APPROVED** [review]

**Review by k15r — APPROVED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**